src/qt/optionsdialog.cpp

```diff
@@ -98,15 +98,39 @@ OptionsDialog::OptionsDialog(QWidget *parent, bool enableWallet) :
     {
         QLocale locale(langStr);
 
-        /** check if the locale name consists of 2 parts (language_country) */
-        if(langStr.contains("_"))
+        if (locale.nativeLanguageName().length()==0) /* If locale name is an empty string, fix it manually. */
         {
-            /** display language strings as "native language - native country (locale name)", e.g. "Deutsch - Deutschland (de)" */
+            if (langStr.contains("szl") && langStr.length()==3)
```

Comment:
instead of this long `if()` statement and repeated code, please make this data-driven, e.g. add a structure
```
/** Additional language names that aren't covered by Qt locale as of [version] */
static const struct {
    const char *abbrev;
    const char *name;
} language_names[] = {
    {"szl", "Silesian"},
    {"pam", "Kapampangan"},
    {"la", "Latin"},
    {"eo", "esperanto"},
};
```